spec/support/shared/with_flag.rb

```diff
@@ -0,0 +1,51 @@
+#  OpenProject is an open source project management software.
+#  Copyright (C) 2010-2022 the OpenProject GmbH
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License version 3.
+#
+#  OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+#  Copyright (C) 2006-2013 Jean-Philippe Lang
+#  Copyright (C) 2010-2013 the ChiliProject Team
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#  See COPYRIGHT and LICENSE files for more details.
+
+module WithFlagMixin
+  module_function
+
+  def with_flags(flags)
+    flags.each do |k, value|
+      name = k.end_with?('?') ? k : "#{k}?"
+
+      raise "#{k} is not a valid flag" unless OpenProject::FeatureDecisions.respond_to?(name)
+
+      allow(OpenProject::FeatureDecisions).to receive(name).and_return value
+    end
+  end
+end
+
+RSpec.configure do |config|
+  config.include WithFlagMixin
+
+  config.before do |example|
+    flags = example.metadata[:with_flag]
+
+    if flags.present?
```

Comment:
`config.before` accepts a filtering option that can be used instead of testing the metadata presence. See https://relishapp.com/rspec/rspec-core/v/3-10/docs/hooks/filters#filtering-hooks-using-symbols

Code would then become
```ruby
  config.before :example, :with_flag do |example|
    with_flags(example.metadata[:with_flag])
  end
```

Also I wrote another rspec helper to enable/disable storages module in specs `modules/storages/spec/support/enable_storages_module.rb`. Maybe it should be replaced by the `with_flag` helper you introduced.

Reply:
Ah, thanks for the pointer, will do.

Reply:
Done in [63bd176](https://github.com/opf/openproject/pull/10799/commits/63bd176a8c03db3de1593244feb8c748fe896e48)

Reply:
You replaced `:enable_storages`. There is also the `:disable_storages` that needs to be replaced by `with_flag: { storages_module_active: false }`.

Reply:
Done